tangelo/algorithms/classical/fci_solver.py

```diff
@@ -131,3 +145,147 @@ def get_rdm(self):
+                                  'qc_module': 'detci', 'fci': True, 'reference': ref})
+
+        # Copy reference wavefunction and swap orbitals to obtain correct active space if necessary
+        wfn = self.backend.core.Wavefunction(self.molecule.solver.mol_nosym, self.molecule.solver.wfn.basisset())
+        wfn.deep_copy(self.molecule.solver.wfn)
+        if n_frozen_occ or n_frozen_vir:
+            mo_order = self.molecule.frozen_occupied + self.molecule.active_occupied + self.molecule.active_virtual + self.molecule.frozen_virtual
+            # Obtain swap operations that will take the unordered list back to ordered with the correct active space in the middle.
+            swap_ops = Permutation(mo_order).transpositions()
+            for swap_op in swap_ops[::-1]:
+                wfn.Ca().rotate_columns(0, swap_op[0], swap_op[1], np.deg2rad(90))
+
+        energy, self.ciwfn = self.backend.energy('fci', molecule=self.molecule.solver.mol,
+                                                 basis=self.basis, return_wfn=True,
+                                                 ref_wfn=wfn)
+        return energy
+
+    def get_rdm(self):
+        """Compute the Full CI 1- and 2-particle reduced density matrices.
+
+        Returns:
+            numpy.array: One-particle RDM.
+            numpy.array: Two-particle RDM.
+
+        Raises:
+            RuntimeError: If method "simulate" hasn't been run.
+        """
+
+        # Check if Full CI has been performed
+        if self.ciwfn is None:
+            raise RuntimeError("FCISolver: Cannot retrieve RDM. Please run the 'simulate' method first")
+        if self.degenerate_mo_energies:
+            warnings.warn("Degenerate orbitals are present in the molecule. The fci calculation is performed "
+                          "without symmetry so the rdms may not correspond to the integrals in molecule. A c1 "
+                          "version of the molecule with the correct integrals is present as FCISolverPsi4.molecule.")
+        one_rdm = np.asarray(self.ciwfn.get_opdm(0, 0, "SUM", False))
+        two_rdm = np.asarray(self.ciwfn.get_tpdm("SUM", True))
+
+        return one_rdm, two_rdm
+
+
+fci_solver_dict = {"pyscf": FCISolverPySCF, 'psi4': FCISolverPsi4}
+
+
+def get_fci_solver(molecule: SecondQuantizedMolecule, solver: Union[None, str, Type[ElectronicStructureSolver]] = default_fci_solver, **kwargs):
+    """Return requested target FCISolver object.
+
+    Args:
+        molecule (SecondQuantizedMolecule) : Molecule
+        solver (string or Type[ElectronicStructureSolver] or None): Supported string identifiers can be found in
+            fci_solver_dict (from tangelo.algorithms.classical.fci_solver). Can also provide a user-defined backend (child to ElectronicStructureSolver class)
+        kwargs: Other arguments that could be passed to a target. Examples are solver type etc.
```

Comment:
Still having trouble here: any concrete example ?

Instead of kwargs, I suggest you name them solver_kwargs. What is a "target" here is not obvious for readers, and the body of your function show that current they are just passed to your solver, so it seems more accurate.